src/differentials/abstract_differential.jl

```diff
@@ -11,18 +11,23 @@ All subtypes of `AbstractDifferential` implement the following operations:
 
 `+(a, b)`: linearly combine differential `a` and differential `b`
 
-`*(a, b)`: multiply the differential `a` by the differential `b`
+`*(a, b)`: multiply the differential `b` by the scaling factor `b`
```

Comment:
This is confusing, to me the call `*(a,b)` would read as multiplying the diff `a` by a factor of `b`

Reply:
yeah thats fair, multiplication of differentiaks and scales is commutative though.

```suggestion
`*(s, d)=*(d, s)`: scale the differential `d` by the scalar `s` (`s` may subtype `AbstractDifferential`)
```

Reply:
^written like that (`*(s, d)=*(d, s)`) i find even more confusing

Reply:
Sorry the confusing part here is we're saying that `*(a,b)` is multiplying the diff `b` by a factor of `b`.

What's happening to `a` here, it's not being used?

Reply:
Oh the old code was wrong.

```suggestion
`*(a, b)`: multiply the differential `b` by the scaling factor `a`
```